server/server.js

```diff
@@ -142,24 +146,49 @@ Server.prototype._handleClientMessage = function(client, data) {
     return;
   }
 
-  logging.info('#client.message - received', client.id, message,
-     (this.resources[message.to] ? 'exists' : 'not instantiated'),
-     (this.subs[message.to] ? 'is subscribed' : 'not subscribed')
-    );
+  if (!this._clientDataPersist(client.id, message)) {
```

Comment:
Here, instead of passing the id, it's better to pass the Client (the new one) object down, so we can just do client.dataStore().

Also, this is why we probably need to rename the existing client to socket first. This way, there is no confusion between the existing cilent (which is the socket) and the new client, which you are currently adding.